packages/ERTP/src/paymentLedger.js

```diff
@@ -20,14 +20,33 @@ import '@agoric/store/exported';
  * @param {Brand} brand
  * @param {AssetKind} assetKind
  * @param {DisplayInfo} displayInfo
+ * @param {((reason: Error) => void)=} optTerminateWithFailure
  * @returns {{ issuer: Issuer, mint: Mint }}
  */
 export const makePaymentLedger = (
   allegedName,
   brand,
   assetKind,
   displayInfo,
+  optTerminateWithFailure = undefined,
 ) => {
+  const destroyLedgerWithFailure = reason => {
+    // TODO destroy ledger state.
```

Comment:
Will this be captured as a Github issue for later, or is this still to be done in this PR?

Seems like it's easy to destroy the `paymentLedger` by setting the variable to `undefined` or another `weakStore`, but we would also need to destroy purses, which carry their own balances as internal state. And we would have to prevent any new minting of payments.

Reply:
Good point about the purses. I'd missed that. Given that, I'll file this as an issue rather than address in this PR.

Reply:
https://github.com/Agoric/agoric-sdk/issues/3434